PR/CE-297-consumir-api-admincore

nicolencinas opened the pull request.

nicolencinas committed:
Eliminando popUpComponente y creando addGrade component

nicolencinas committed:
modificando nombres de los componentes

nicolencinas committed:
subiendo modificaciones de vulnerabilidades de sonar TSliny

nicolencinas committed:
commit de backup

nicolencinas committed:
subiendo correcciones parciales de vulnerabilidades de sonar

nicolencinas committed:
subiendo cambios

oscarumbert34 committed:
merge

nicolencinas committed:
subiendo camvios

nicolencinas committed:
Merge branch 'feature/CE-295-emprolijar-codigo-parte-1' of https://github.com/oscarumbert/clickEscuela into feature/CE-295-emprolijar-codigo-parte-1

oscarumbert34 committed:
delete import

oscarumbert34 committed:
Merge branch 'develop' into feature/CE-295-emprolijar-codigo-parte-1

nicolencinas committed:
corrigiendo la gran mayoria de vulnerabilidades de sona

nicolencinas committed:
subiendo consumo de admin core GET

nicolencinas committed:
cambiando animacion de spinner por logo de click escuela

nicolencinas committed:
Agregando Post de students,

Se mejora codigo para inyectar svg, y se pasa el snack bar a un servicio para unificar los mensajes con sus estilos

nicolencinas committed:
Se crean constantes para separar los ngmodels del codigo principal

nicolencinas committed:
mejorando codigo

clickescuela commented on src/app/services/student.service.ts:
```diff
@@ -229,11 +230,17 @@ export class studentService {
   // Aca incia el codigo para consumo de api
 
-  getStudents(fulldetail: boolean): Observable<any> {
-    const path = 'https://click-admin.herokuapp.com/click-escuela/admin-core/school/1365/student?fullDetail=' + fulldetail;
```
this url should be in file environments.ts of angular

clickescuela commented on src/app/components/admin/students-component/add-student/add-student.component.ts:
```diff
@@ -1,11 +1,13 @@
 import { Component, OnInit } from '@angular/core';
-import { Student } from 'src/app/models/student';
-import { Parent } from 'src/app/models/parent';
+// import { Student } from 'src/app/models/student';
```
delete code commented

clickescuela commented on src/app/components/admin/students-component/add-student/add-student.component.ts:
```diff
@@ -114,29 +117,45 @@ export class AddStudentComponent implements OnInit {
+    console.log(this.currentStudent);
+    this.studentsService.addStudentPost(this.currentStudent, this.schoolId).subscribe(
+      data => {
+        this.snackBarService.showSnackBar('El alumno se creo satisfactoriamente','Aceptar', 'SUCCES');
```
Messages must be Constants

clickescuela commented on src/app/components/admin/students-component/add-student/add-student.component.ts:
```diff
@@ -114,29 +117,45 @@ export class AddStudentComponent implements OnInit {
   cancelAdd() {
     this.resetStudentModel();
-    this.showSnackBar('Se limpiaron los formularios');
+    this.snackBarService.showSnackBar('Se limpiaron los formularios','Aceptar', 'NORMAL');
```
This message must be a constant

clickescuela commented on src/app/components/admin/students-component/student-base-model/student-base-model.component.ts:
```diff
@@ -69,6 +78,7 @@ export class StudentBaseModelComponent implements OnInit {
 
   ngOnInit() {
 
+    console.log(this.data)
```
delete this line

clickescuela commented on src/app/components/admin/students-component/student-base-model/student-base-model.component.ts:
```diff
@@ -121,21 +131,21 @@ export class StudentBaseModelComponent implements OnInit {
       data => {
         this.dataSource.data = data;
         if (JSON.stringify(this.dataSource.data) === JSON.stringify(this.studentsArray)) {
-          this.snackbar.open('Recarga exitosa. No se encontraron nuevas entradas', 'Aceptar', { duration: 3000 });
```
This message must be a constant

clickescuela commented on src/app/components/admin/students-component/student-base-model/student-base-model.component.ts:
```diff
@@ -121,21 +131,21 @@ export class StudentBaseModelComponent implements OnInit {
         this.dataSource.data = data;
         if (JSON.stringify(this.dataSource.data) === JSON.stringify(this.studentsArray)) {
-          this.snackbar.open('Recarga exitosa. No se encontraron nuevas entradas', 'Aceptar', { duration: 3000 });
+          this.snackbar.showSnackBar('Recarga exitosa. No se encontraron nuevas entradas', 'Aceptar','NORMAL');
```
This message must be a constant

clickescuela commented on src/app/components/admin/students-component/student-base-model/student-base-model.component.ts:
```diff
@@ -121,21 +131,21 @@ export class StudentBaseModelComponent implements OnInit {
       },
       err => {
         console.log(err);
-        this.snackbar.open('Se produjo un error.', 'Aceptar', { duration: 3000 });
```
This message must be a constant

clickescuela commented on src/app/services/student.service.ts:
```diff
@@ -229,11 +230,17 @@ export class studentService {
-  getStudents(fulldetail: boolean): Observable<any> {
-    const path = 'https://click-admin.herokuapp.com/click-escuela/admin-core/school/1365/student?fullDetail=' + fulldetail;
+  getStudents(fulldetail: boolean, idSchool: string): Observable<any> {
+    const path = 'https://click-admin.herokuapp.com/click-escuela/admin-core/school/' + idSchool + '/student?fullDetail=' + fulldetail;
```
this url should be in file environments.ts of angular

clickescuela commented on src/app/services/student.service.ts:
```diff
@@ -229,11 +230,17 @@ export class studentService {
     return this.connector.get<any>(path);
   }
 
+  addStudentPost(student: StudentI, idSchool: string): Observable<StudentI> {
```
this url should be in file environments.ts of angular

clickescuela commented on src/app/components/admin/students-component/add-student/add-student.component.ts:
```diff
@@ -117,18 +81,18 @@ export class AddStudentComponent implements OnInit {
   addParent() {
     this.secondParent = !this.secondParent;
     this.secondParent
-      ? this.snackBarService.showSnackBar('Se agrego un familiar','Aceptar', 'NORMAL')
```
Messages must be variables constant

clickescuela commented on src/app/components/admin/students-component/add-student/add-student.component.ts:
```diff
@@ -146,7 +110,7 @@ export class AddStudentComponent implements OnInit {
   cancelAdd() {
     this.resetStudentModel();
-    this.snackBarService.showSnackBar('Se limpiaron los formularios','Aceptar', 'NORMAL');
+    this.snackBarService.showSnackBar('Se limpiaron los formularios', 'Aceptar', 'NORMAL');
```
Messages must be variables constant

nicolencinas committed:
subiendo load-screen component

nicolencinas committed:
subiendo cambios en los comentarios de pr

nicolencinas committed:
subiendo cabios

nicolencinas committed:
Se quita boton de agregado de eventos al calendario

Se quita el botón de nuevo evento por ser poco intuitivo, se soluciona error anterior que abria el event-detail y el add-event con el stopPropagation

nicolencinas committed:
subiendo editado de alumnos

nicolencinas committed:
subiendo cambios

nicolencinas committed:
subiendo cambios de animaciones

nicolencinas committed:
cambiando animaciones de carga

nicolencinas committed:
Merge branch 'develop' into feature/CE-297-consumir-api-admincore
